plugins/helo

```diff
@@ -345,7 +345,7 @@ sub invalid_localhost {
         $self->log(LOGDEBUG, "pass, is localhost");
         return;
     }
-    if ($host && lc $host eq 'localhost') {
+    if ($host && lc $host ne 'localhost') {
         $self->log(LOGDEBUG, "pass, host is localhost");
```

Comment:
Pseudocode before change:

If remote IP is localhost, then pass.
If HELO is localhost (and implicitly: remote IP is NOT localhost), then pass. (this is in fact the situation we want to be rejecting)
Otherwise deny.

After change:

If remote IP is localhost, then pass.
If HELO is NOT localhost (and implicitly: remote IP is NOT localhost), then pass.
Otherwise deny.

I'm sure my version makes more sense. I just forgot to change the LOGDEBUG message, and the fact I botched my pull request (should have been 2 different pullreqs). I can redo this in a few hours you agree with me.